src/libraries/System.Runtime.Extensions/tests/System/Random.cs

```diff
@@ -753,7 +753,41 @@ public static void Shuffle_Span_Seeded(bool emptyShuffle)
             }
 
             random.Shuffle(items);
-            AssertExtensions.SequenceEqual(stackalloc int[] { 1, 4, 3, 2 }, items);
+            AssertExtensions.SequenceEqual(stackalloc int[] { 2, 1, 4, 3 }, items);
+        }
+
+        [Fact]
+        public static void Shuffle_Array_Fairness()
+        {
+            // Test that repeatedly shuffling an array puts each value in each position a roughly equal number of times.
+            // This does not prove that the shuffle is fair but will at least fail if the shuffle is obviously unfair
+            const int runs = 10000;
+            const int len = 50;
+            int[,] counts = new int[len, len];
+            var rng = new Random(123);
+            for (int i = 0; i < runs; i++)
+            {
+                int[]? array = new int[len];
+                for (int index = 0; index < len; index++)
+                {
+                    // Fill the array with the numbers 0..len
+                    array[index] = index;
+                }
+
+                rng.Shuffle(array);
+
+                for (int j = 0; j < len; j++)
+                {
+                    int index = array[j];
+                    counts[j,index] += 1;
+                }
+            }
+    
+            const int expected = runs / len;
+            foreach (int count in counts)
+            {
+                Assert.InRange(count, expected / 10 * 7, expected * 13 / 10);
```

Comment:
where do these numbers come from? can they be calculated with some heuristic?

Reply:
eh, I guess if +/-30% is stable in practice, we don't need to do the math.

Although, on the fancier end, I was going to add a Chi-sq test for Random --
https://gist.github.com/danmoseley/784a25839978425ae15d037f587ac76b

Reply:
Great idea - I've switched to chi-squared for this.